crates/bitwarden-vault/src/cipher/cipher_permissions.rs

```diff
@@ -0,0 +1,10 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Copy, Deserialize, Debug, JsonSchema, Clone, PartialEq)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+pub struct CipherPermissions {
+    pub delete: bool,
+    pub restore: bool,
```

Comment:
@JimmyVo16 I haven't followed this work, but any reason we don't move edit and view password into this struct and deprecated the current location? We'd have to continue sending both for a while but could eventually have all server side permissions here?